### ISSUE TITLE
ML Week 2 project review

### ISSUE THREAD

**Issue by bonareri** [Tactevil/project-blueberry#1]
<html>
<body>
<html><head></head><body><h1>Week 2 ML Project Feedback</h1><p><strong>Score: 88/125 — Good</strong></p><p>You put good effort into this project and demonstrated strong initiative by exploring feature engineering, feature selection, hyperparameter tuning, overfitting checks, feature importance, and ensemble modelling. Your overall workflow shows that you are becoming more comfortable with the machine learning process.</p><h2>What You Did Well</h2><ul><li><p><strong>EDA:</strong> Thoroughly explored the dataset, including missing values, duplicates, descriptive statistics, distributions, and correlations.</p></li><li><p><strong>Target analysis:</strong> Examined the <code inline="">yield</code> distribution and its skewness.</p></li><li><p><strong>Preprocessing:</strong> Correctly separated the target and <code inline="">id</code>, and handled scaling using the training data.</p></li><li><p><strong>Validation:</strong> Created a separate validation set to evaluate model performance.</p></li><li><p><strong>Feature engineering:</strong> Created useful temperature, bee-population, and rainfall-related features.</p></li><li><p><strong>Feature selection:</strong> Experimented with correlation and mutual information.</p></li><li><p><strong>Hyperparameter tuning:</strong> Used <code inline="">RandomizedSearchCV</code> to improve the Random Forest model.</p></li><li><p><strong>Model diagnostics:</strong> Compared training and validation performance to check for overfitting.</p></li><li><p><strong>Feature importance:</strong> Investigated which features were most influential in the model.</p></li><li><p><strong>Submission:</strong> Successfully generated predictions in the required <code inline="">id</code> and <code inline="">yield</code> format.</p></li><li><p><strong>Reproducibility:</strong> Used <code inline="">random_state</code> consistently.</p></li></ul><h2>Areas for Improvement</h2><h3>1. Complete the Required Models</h3><p>The project required:</p><ul><li><p>Linear Regression</p></li><li><p>Polynomial Regression</p></li><li><p>Ridge</p></li><li><p>Lasso</p></li><li><p>Elastic Net</p></li></ul><p>You only implemented <strong>Ridge and Lasso</strong> from the required models, while focusing mainly on Random Forest and Gradient Boosting.</p><p>The advanced models are useful, but they should come <strong>after</strong> completing and comparing the five required models.</p><h3>2. Use MAE for Model Selection</h3><p>The competition metric is <strong>MAE</strong>, but your cross-validation and model selection were based primarily on <strong>R²</strong>.</p><p>MAE should be your main metric because it directly reflects the competition objective. R² can still be reported as a supplementary metric.</p><h3>3. Make Your Model Selection Consistent</h3><p>You identified <strong>Gradient Boosting as the best baseline model based on R²</strong>, but then proceeded to tune Random Forest.</p><p>If Gradient Boosting is your best model, you should either tune it or clearly explain why you chose to tune Random Forest instead.</p><h3>4. Fix the Feature Selection Logic</h3><p>Your explanation says you are taking the <strong>intersection</strong> of correlation-based and mutual-information features. However, the code falls back to the top 15 mutual-information features when the intersection is too small.</p><p>Make sure your explanation accurately reflects what the code is actually doing.</p><h3>5. Improve Feature Engineering</h3><p>You created several interesting features, but the reasoning behind them was not clearly explained.</p><p>For each engineered feature, explain the hypothesis behind creating it and why you expect it to help predict yield.</p><p>Also, consider creating the engineered features before scaling so that the relationships remain easier to interpret.</p><h3>6. Evaluate the Ensemble Using MAE</h3><p>You selected the ensemble because its R² improved from 0.8217 to 0.8253.</p><p>However, a higher R² does not necessarily mean a lower MAE. Since MAE is the competition metric, you should compare the ensemble's <strong>MAE</strong> with the individual models before selecting it as the final model.</p><h3>7. Compare Models More Systematically</h3><p>Your final comparison should clearly show how all required models perform using the same evaluation approach.</p><h2> Overall Feedback</h2><p>This was a <strong>technically ambitious submission</strong>. You went beyond the basic requirements and demonstrated good understanding of several important ML concepts, particularly feature engineering, model tuning, validation, and model interpretation.</p><p>The main issue is <strong>focus and alignment with the project requirements</strong>. You moved into more advanced models before fully implementing and comparing the five models that were specifically required for this week's work.</p><p>Going forward, focus on building a strong baseline first: <strong>implement all required models, evaluate them using MAE, tune them systematically, and then experiment with advanced models.</strong> Also make sure that your written conclusions match exactly what your code is doing.</p><blockquote><p><strong>The goal is not to use the most advanced model, but to build a well-justified modelling process that uses the right models, metrics, and evaluation strategy.</strong></p></blockquote><p><strong>Final Score: 88/125 — Good</strong></p></body></html>
</body>
</html>
